Degrade missing VFS overlays in serialized Swift options to warnings.

adrian-prantl opened the pull request:
The VFS options turn into fatal errors when the referenced file is not found. Since the Xcode build system tends to create a lot of VFS overlays by default, stat them emit a warning if the yaml file couldn't be found.

rdar://112939483

adrian-prantl commented:
@swift-ci test

JDevlieghere commented on lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:
```diff
@@ -1539,6 +1541,26 @@ void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string>& source,
 
+    // The VFS options turn into fatal errors when not found. Since
+    // the Xcode build system tends to create a lot of VFS overlays by
+    // default, stat them emit a warning if the yaml file couldn't be found.
```
`s/stat them emit/stat them and emit/`

JDevlieghere reviewed (approved):
LGTM

adrian-prantl committed:
Move output of debugger warnings into SwiftASTContext.

The only noticeable side effect is that now each diagnostics printed separately.

adrian-prantl committed:
Degrade missing VFS overlays in serialized Swift options to warnings.

The VFS options turn into fatal errors when the referenced file
is not found. Since the Xcode build system tends to create a
lot of VFS overlays by default, stat them emit a warning if the
yaml file couldn't be found.

rdar://112939483

adrian-prantl commented:
Changed:
- The previous patch did the filtering before applying path remappings, now it's done after.
- Each warning is now emitted separately

adrian-prantl commented:
@swift-ci test

adrian-prantl commented:
@swift-ci test windows
